### ISSUE TITLE
Heavy Fracking drill

### ISSUE THREAD

**Issue by skyraze** [nukeguard/NaniteConstructionSystem#47]
so i just had an idea pop in my head while i was falling asleep, I was thinking along with the NUHOL miner, have a Heavy Fracking drill.
a massive drill that is 4X4X10(or similar height) large that can find and source out materials within a 3KM radius. it would require massive amounts of materials including ice and platinum, so you need to go into space first before you can use it on a planet.

it's a high maintenance, high payout device that needs Stone, Ice, and massive amounts of power (500MW).
to get the idea, Ice is used to cool and lubricate the device, stone is used in conjunction with the ice to create fracturing fluid. then nanites do the rest of the work and grabs and return the ores. (This is the concept idea to make the drill make sense, just letting it detect ore and having the nanites grab it is enough, plus lowers the amount of CPU load needed to drill actual bores)

i'm still contemplating whether the drill will supply its own nanites or if you need a factory to do that.
I feel like needing factories to supply nanites is the way to go, that way you can have seperate ships use the drill wirelessly instead of having to connect to the station.

if it's possible, make it be able to only source out one pocket of ore at a time.
so if it finds a silver ore pocket and a gold ore pocket, you have to choose one or the other and not both at the same time. That also means if it finds two gold ore pockets, you have to choose between the two, not both together. that way the Overpowered nature of it is balanced by maintenance, power cost, and filtering.

**Comment by Fank** [comment]
@nukeguard I think this could be named mining rig

**Comment by skyraze** [comment]
https://i.pinimg.com/564x/12/17/a5/1217a52f9d22a3545634b7ab2e9165f8.jpg
https://upload.wikimedia.org/wikipedia/commons/5/5d/BarnettShaleDrilling-9323.jpg

first image is very close to the model i was thinking of.
second image shows what a current drill looks like.

**Comment by Fank** [comment]
@skyraze i updated your first one to the page with high res, pictures.

**Comment by Fank** [comment]
The model is similar to the Half Life 2 Combine Restrictor (Thumper)  https://www.youtube.com/watch?v=99bEgHkM3Lw

**Comment by skyraze** [comment]
quite a similar design, but meant to have a laser drill like the concept for the nuhol miner.

**Comment by Fank** [comment]
Hm why not thinking about a bigger version? Like:
![image](https://user-images.githubusercontent.com/1900106/44051396-0859574c-9f3a-11e8-9928-d8a6481faf68.png)

Every color is a block:
- Green (just surface, ignore it)
- Yellow (Main block, tower/computing base)
- Blue (Scanning modules, act as upgrade module)
- Red (Antenna module, act as upgrade module)

All blocks are required to make the main block work. One single block isn't enought, for such project it should require multiple parts to work. In my example above the main block is 5x5x3 big and high maybe should be 5x5x10. The scanning modules are  5x4x2 long and flat. and the antenna is 3x3x4 should be small and high.

Because 3km range is **** huge, example of easy earth starter base with model of above in cursor:
![image](https://user-images.githubusercontent.com/1900106/44051936-83536644-9f3b-11e8-9955-6bf52367260c.png)

**Comment by skyraze** [comment]
I was actually thinking about, and it would work perfectly with the 2.0+
updates. They are extra pieces that attach to the module points on the
current miner, turns it from the 3X3X4 design to a 5X5x10 permanent drill
platform.

**Comment by Fank** [comment]
No, what i mean a total dimension about 15x15x25, because the upgrades does not act as upgrades there are the required parts to the whole. But using a single block for it would be too easy.

**Comment by skyraze** [comment]
Pretty much what I was thinking, you need each module else the whole thing won't work.
But the base is the current miner that @nukeguard is making, then you need 4 blocks, one on each side of that miner that covers all the module points,  then place a top to the miner and that finishes the fracking drill. I'll put together a visual representation of what I'm thinking, I'll bring your ideas into it too because I like it.

**Comment by Fank** [comment]
And No, the model which is @nukeguard currently working on i a standalone model, and its an ore detector not a miner. all blocks should build the model, every single block should not make sence. Its more the idea to build such a big block step by step without allocation the space volume for it.
So each block should looks stupid, untill you build the whole bock combination for the complete one, that should look awesome.

**Comment by Fank** [comment]
And i think its the best that the main block has its only nanite system for mining, due the range it can have.
Control facilites are limited to 300m or 500m not sure right now. Maybe same as the current mining works like a control facility contact the main block, so nanites will travel to the antenna of it and then the main block will start spreading its own nanites to collect ore and will transfer them to the nanites which waits on the antenna and than they will return to the control facility.
In the current planned version with 8 military grade speed upgrades on a control facility nanites can reach a speed of 200m/s, but normal level 1 upgrades its about 56m/s i think. I think the mining nanite speed of the main block should be around 500m/s.

**Comment by Fank** [comment]
So what we got for now, because the drill system does not fit on the nanite system.
We will use drill nanites for it, these are large than the normal ones, drill tunnels while flying, and slower, and have a "cargo" which means if they are full they return.
Upgrades for them could be:
- More drill nanites
- Faster drill nanites
- Storage upgrade so drill nanites can carry more.

**Comment by Fank** [comment]
And yes the drill nanites should use a tunnel system for flying, so if they started drilling they will first travel down the x axis then will travel to the y axis and then the z axis until target reached.
If they reach the ore, it will start mining it like a cube.

**Comment by nukeguard** [comment]
I can convert the model i have now to make it the basic mining facility, that you add the massive upgrades to and make the top mountable for the top upgrade
top upgrade would be the range extension 3km, and the four double upgrade ports are for the arms that provide the other fuctions.

**Comment by Fank** [comment]
As much as i hope, but i think i need to close this one. Because checking voxels with a radius of more than 200 meters is nearly killing the server:

![image](https://user-images.githubusercontent.com/1900106/44285341-3d618d00-a265-11e8-9ada-a897c4aa1955.png)

If there is someone who finds a solution how we can read voxel information over a range of 200m, without using more than 10% usage, than we can continue on this.

**Comment by skyraze** [comment]
doesn't the ore detector do around 500M? unless i'm missing something.
